examples/client_objects.rs

```diff
@@ -0,0 +1,51 @@
+use azure_identity::{
+    ManagedIdentityCredential,
+    ManagedIdentityCredentialOptions,
+    UserAssignedId
+};
+use azure_security_keyvault_secrets::{
+    SecretClient, 
+    SecretClientOptions
+};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+
+    let user_assigned_id_value = std::env::var("AZURE_USER_ASSIGNED_IDENTITY")
+        .map_err(|_| {
+            std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "AZURE_USER_ASSIGNED_IDENTITY environment variable is required",
+            )
+        })?;
+
+    let user_assigned_id = UserAssignedId::ClientId(user_assigned_id_value);
+
+    let credential_options = ManagedIdentityCredentialOptions {
+        user_assigned_id: Some(user_assigned_id),
+        ..Default::default()
+    };
+
+    let credential = ManagedIdentityCredential::new(Some(credential_options))?;
+
+    let key_vault_endpoint = std::env::var("AZURE_KEY_VAULT_ENDPOINT")
+        .map_err(|_| {
+            std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "AZURE_KEY_VAULT_ENDPOINT environment variable is required",
+            )
+        })?;
+
+    let secret_client_options = SecretClientOptions {
+        api_version: "7.5".to_string(),
```

Comment:
Is this germane to the example? Customers generally shouldn't do this. Our official support policy is that we support the version we generated for those versions of crates If you upgrade a crate you might get a different api-version. You can specify an older (or newer, but that's worse) version but we can't guarantee it'll work; so crate versions are ~= api-versions.

Reply:
Removed the option entirely